### PR TITLE
Use raw options for keyval handing

### PR DESCRIPTION
**Pull request by josephwright**

### CODE REVIEW

**Comment by josephwright** [comment]
@skillmon Could you take a look here too?

**Review by davidcarlisle — APPROVED** [review]
looks fine I noticed one typo in the comments, that could be picked up before merging, but not essential

**Comment by davidcarlisle** [comment]
test github-0710 is showing a diff but I don't see that locally (general-option-C  = 333 being set or not)

File: base/testfiles/github-0862.lvt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+\documentclass{article}
+
+\begin{filecontents}[force]{\jobname.sty}
+  \ProvidesPackage{\jobname}
+  \DeclareKeys{foo .code = \typeout{Foo was gives: \detokenize{#1}}}
```
Comment (Skillmon) [inline base/testfiles/github-0862.lvt:5]: ~~gives~~ given

**Pull request merged**